src/util/antlr/ANTLRErrorHandling.h

```diff
@@ -4,21 +4,54 @@
 
 #pragma once
 
+#include <parser/ParseException.h>
+
 #include <string>
 
-#include "../../parser/ParseException.h"
 #include "BaseErrorListener.h"
+#include "CharStream.h"
+#include "CommonTokenStream.h"
 #include "DefaultErrorStrategy.h"
+#include "ParserRuleContext.h"
 #include "RecognitionException.h"
+#include "Recognizer.h"
+#include "TokenSource.h"
+
+// TODO<qup42> whats up with this namespace?
+namespace {
+[[maybe_unused]] std::string colorError(const ExceptionMetadata& metadata) {
+  if (metadata.startIndex < metadata.stopIndex) {
+    auto first = metadata.query_.substr(0, metadata.startIndex);
+    auto middle = metadata.query_.substr(
+        metadata.startIndex, metadata.stopIndex - metadata.startIndex + 1);
+    auto end = metadata.query_.substr(metadata.stopIndex + 1);
 
-struct ThrowingErrorStrategy : public antlr4::DefaultErrorStrategy {
-  void reportError(antlr4::Parser*,
-                   const antlr4::RecognitionException& e) override {
-    throw antlr4::ParseCancellationException{
-        e.what() + std::string{" at token \""} +
-        e.getOffendingToken()->getText() + '"'};
+    return (first + "\x1b[1m\x1b[4m\x1b[31m" + middle + "\x1b[0m" + end);
+  } else {
+    return metadata.query_;
```

Comment:
I can think of two cases: 
- The position of the error cannot be determined (then the error message should state something like that).
- The error position is a single position (but isn't then stopIndex = endIndex +1?), In this case this position should be marked.

In general I would not log the complete query, because this is something that the upstream code can also do if we want so.

Reply:
This may happen in the case that the Tokens has no position. Inserted tokens that don't have a position in the input have a index of the constant INVALID_INDEX.

I should also add checks that the first parameter to `substr` is not larger than the string itself. `substr` throws exception in these case.